shared/src/commonMain/kotlin/io/customer/shared/di/KMMComponent.kt

```diff
@@ -0,0 +1,26 @@
+package io.customer.shared.di
+
+import io.customer.shared.Platform
+import io.customer.shared.sdk.config.CustomerIOConfig
+
+/**
+ * Workspace component dependency graph to satisfy workspace based dependencies from single place.
+ *
+ * The class should only contain dependencies matching the following criteria:
+ * - dependencies that requires workspace information to be initialized.
+ * - dependencies that should be reinitialized whenever workspace is switched.
+ * - or any dependency that cannot fit in DIGraphs above this graph :)
+ *
+ * @see [KMMStaticComponent] for complete graph hierarchy.
+ */
+@Suppress("MemberVisibilityCanBePrivate", "unused")
+class KMMComponent(
+    private val staticComponent: KMMStaticComponent = KMMStaticComponent(),
+    val platform: Platform,
+    val sdkConfig: CustomerIOConfig,
+) : DIGraph() {
```

Comment:
Speaking from experience, I am hesitant for us creating multiple separate `DiGraph` instances compared to just having 1 graph with all dependencies in it like we do today in the Android and iOS SDK. 

In the past, I created ~4 separate di graphs in the iOS SDKs. It was very difficult to maintain code and tests for. It got so full of boilerplate and hard to read code that I ended up refactoring the SDK and just going back to 1 graph. 

I would love in the future when we use KMM more in the team, that we have the DiGraph implemented in KMM and the Android and iOS SDK simply extends the 1 DiGraph in shared KMM. All to try and keep 1 DiGraph in the SDK to make things easier to write and maintain.

Reply:
I think it's not a good idea to keep everything behind 1 graph even when the classes are not dependent. I want to keep graphs simple (docs on `KMMStaticComponent` might help understanding the idea behind it). Also, keeping only one graph that is dependent on configurations and other things had made things difficult for us in the past when working on wrapper SDKs as the initialization of SDK can be delayed. Also, I think it might be helpful is setting up test cases as well, but not yet sure.

Can you please share from your experience in iOS that what forced you to revert back to single graph?